types/react/jsx-dev-runtime.d.ts

```diff
@@ -1,2 +1,12 @@
-// Expose `JSX` namespace in `global` namespace
-import './';
+import * as React from './';
+
+export namespace JSX {
```

Comment:
namespaces are not exactly types in TS and have proven to be somewhat difficult to work with here because this ain't possible:
```ts
export type JSX = React.JSX
```

A namespace also cannot be "passed around", I can only unpack a property from it and use that - that's why I've used this "reexporting namespace" pattern here 🤷‍♂️